### DISCUSSION TITLE
How to use nullable values in CustomType with SQL statement

### DISCUSSION THREAD

**Discussion by boan-anbo** [SeaQL/sea-orm#1128] [Q&A]
Hi there! I have a custom type to work with a raw SQL statement.

```rust
#[derive(Debug, FromQueryResult)]
pub struct KeyGroups {
    pub key: String,
    pub value: String,
    pub key_count: i32,
}
```

I can get this to work with SeaOrm query when value is not null`

```rust
let tags = KeyGroups::find_by_statement(
            Statement::from_string(
                DbBackend::Sqlite,
                r#"SELECT key, value, Count(key) as 'key_count' from tag group by tag.key, tag.value"#.to_owned(),
            )
        )
            .all(&db)
            .await
```

However, SeaOrm throws when the value is null: `Query("error occurred while decoding value: Null")`

But if I make the CustomType `value` field nullable, like so:
```rust
pub struct KeyGroups {
    pub key: String,
    pub value: Option<String>, // nullable
    pub key_count: i32,
}
```
It still throws decoding error. 

How do I use nullable fields with CustomType and raw Sql Statement? Many thanks in advance!

**Comment by billy1624** [comment]
Hey @boan-anbo, can you query the SQL `SELECT key, value, Count(key) as 'key_count' from tag group by tag.key, tag.value` and observe the result? I think maybe it's not value column contains NULL value but also key column as well?
